File: create_tables/spatial_description.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table spatial_description';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute(
    """create table spatial_description(
    SPATIAL_DESCRIPTION_ID VARCHAR(40) PRIMARY KEY NOT NULL,
    SPATIAL_OBS_NO INT,
    ACTIVE_IND VARCHAR(1),
    CARTER_IND VARCHAR(1),
    CONGRESS_IND VARCHAR(1),
    COORD_ACQUISITION_ID VARCHAR(40),
    COORD_SYSTEM_ID VARCHAR(40),
    DLS_IND VARCHAR(1),
    EFFECTIVE_DATE TIMESTAMP(0),
    EXPIRY_DATE TIMESTAMP(0),
    FPS_IND VARCHAR(1),
    GEODETIC_IND VARCHAR(1),
    GROSS_SIZE DECIMAL(20,10),
    GROSS_SIZE_OUOM VARCHAR(40),
    INACTIVATION_DATE TIMESTAMP(0),
    LIBYA_IND VARCHAR(1),
    LINE_IND VARCHAR(1),
    LINE_VERSION_IND VARCHAR(1),
    LOCAL_COORD_SYSTEM_ID VARCHAR(40),
    MAX_LATITUDE DECIMAL(14,9),
    MAX_LONGITUDE DECIMAL(14,9),
    MINERAL_ZONE_IND VARCHAR(1),
    MIN_LATITUDE DECIMAL(14,9),
    MIN_LONGITUDE DECIMAL(14,9),
    NE_LOC_IND VARCHAR(1),
    NORTH_SEA_IND VARCHAR(1),
    NTS_IND VARCHAR(1),
    OFFSHORE_IND VARCHAR(1),
    OHIO_IND VARCHAR(1),
    PBL_IND VARCHAR(1),
    POINT_IND VARCHAR(1),
    POINT_VERSION_IND VARCHAR(1),
    POLYGON_IND VARCHAR(1),
    POLYGON_VERSION_IND VARCHAR(1),
    PPDM_GUID VARCHAR(38),
    REFERENCE_NUM VARCHAR(30),
    REMARK VARCHAR(2000),
    SOURCE VARCHAR(40),
    SPATIAL_DESC_TEXT_IND VARCHAR(1),
    SPATIAL_DESC_TYPE VARCHAR(40),
    TEXAS_IND VARCHAR(1),
    ROW_CHANGED_BY VARCHAR(30),
    ROW_CHANGED_DATE TIMESTAMP(0),
    ROW_CREATED_BY VARCHAR(30),
    ROW_CREATED_DATE TIMESTAMP(0),
    ROW_EFFECTIVE_DATE TIMESTAMP(0),
    ROW_EXPIRY_DATE TIMESTAMP(0),
    ROW_QUALITY VARCHAR(40))""")

File: postgres_39/struct/SpatialDescription.go
package dto

import (
	"time"
)

type Spatial_description struct {
	Spatial_description_id string     `json:"spatial_description_id"`
	Spatial_obs_no         int        `json:"spatial_obs_no"`
	Active_ind             *string    `json:"active_ind"`
	Carter_ind             *string    `json:"carter_ind"`
	Congress_ind           *string    `json:"congress_ind"`
	Coord_acquisition_id   *string    `json:"coord_acquisition_id"`
	Coord_system_id        *string    `json:"coord_system_id"`
	Dls_ind                *string    `json:"dls_ind"`
	Effective_date         *time.Time `json:"effective_date"`
	Expiry_date            *time.Time `json:"expiry_date"`
	Fps_ind                *string    `json:"fps_ind"`
	Geodetic_ind           *string    `json:"geodetic_ind"`
	Gross_size             *float64   `json:"gross_size"`
	Gross_size_ouom        *string    `json:"gross_size_ouom"`
	Inactivation_date      *time.Time `json:"inactivation_date"`
	Libya_ind              *string    `json:"libya_ind"`
	Line_ind               *string    `json:"line_ind"`
	Line_version_ind       *string    `json:"line_version_ind"`
	Local_coord_system_id  *string    `json:"local_coord_system_id"`
	Max_latitude           *float64   `json:"max_latitude"`
	Max_longitude          *float64   `json:"max_longitude"`
	Mineral_zone_ind       *string    `json:"mineral_zone_ind"`
	Min_latitude           *float64   `json:"min_latitude"`
	Min_longitude          *float64   `json:"min_longitude"`
	Ne_loc_ind             *string    `json:"ne_loc_ind"`
	North_sea_ind          *string    `json:"north_sea_ind"`
	Nts_ind                *string    `json:"nts_ind"`
	Offshore_ind           *string    `json:"offshore_ind"`
	Ohio_ind               *string    `json:"ohio_ind"`
	Pbl_ind                *string    `json:"pbl_ind"`
	Point_ind              *string    `json:"point_ind"`
	Point_version_ind      *string    `json:"point_version_ind"`
	Polygon_ind            *string    `json:"polygon_ind"`
	Polygon_version_ind    *string    `json:"polygon_version_ind"`
	Ppdm_guid              string     `json:"ppdm_guid"`
	Reference_num          *string    `json:"reference_num"`
	Remark                 *string    `json:"remark"`
	Source                 *string    `json:"source"`
	Spatial_desc_text_ind  *string    `json:"spatial_desc_text_ind"`
	Spatial_desc_type      *string    `json:"spatial_desc_type"`
	Texas_ind              *string    `json:"texas_ind"`
	Row_changed_by         *string    `json:"row_changed_by"`
	Row_changed_date       *time.Time `json:"row_changed_date"`
	Row_created_by         *string    `json:"row_created_by"`
	Row_created_date       *time.Time `json:"row_created_date"`
	Row_effective_date     *time.Time `json:"row_effective_date"`
	Row_expiry_date        *time.Time `json:"row_expiry_date"`
	Row_quality            *string    `json:"row_quality"`
}
